engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/CreateProcessInstanceProcessor.java

```diff
@@ -79,8 +89,12 @@ public boolean onCommand(
     }
 
     final var processInstance = initProcessInstanceRecord(process, processInstanceKey);
-    commandWriter.appendFollowUpCommand(
-        processInstanceKey, ProcessInstanceIntent.ACTIVATE_ELEMENT, processInstance);
+    if (record.startInstructions().isEmpty()) {
+      commandWriter.appendFollowUpCommand(
+          processInstanceKey, ProcessInstanceIntent.ACTIVATE_ELEMENT, processInstance);
+    } else {
+      activateElementsForStartInstructions(record, process, processInstanceKey, processInstance);
```

Comment:
:wrench: You could pass in just the start instructions. The record is only used to get to the start instructions